File: pset6/caesar/caesar.py
from cs50 import get_string
import sys
# checks for the exactly 2 arguments
if len(sys.argv) != 2:
    print('Nope')
    exit
# get the key value to cipher the plaintext
key = int(sys.argv[1])
# if key is less than 0
if key < 0:
    print("Nope", end = "")
    exit

print ("plaintext: ", end= "")
# getting the plain text
code = get_string()

print("ciphertext: ", end = "")
# loop through the plaintext
for char in code:
    if char.islower():
        print(chr((((ord(char) + key) -97 ) % 26) + 97), end="")
    elif char.isupper():
        print(chr((((ord(char) + key) -65 ) % 26) + 65), end="")
    else:
        print(char, end="")

print()

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>


 int main(int argc, string argv[])
 {

    int key=atoi(argv[1]);
    if (argc != 2)
    {
        printf("Nope\n");
        return 1;
    }

    if (key < 0)
    {
        printf("Nope\n");
        return 1;
    }
    else
    {

        string code = get_string("plaintext: ");
        printf("\nciphertext: ");
        for (int i = 0, n=strlen(code); i<n;i++)
            {
                if islower(code[i])
                    printf("%c", (((code[i] + key) - 97) % 26) + 97);
                else if isupper(code[i])
                    printf("%c", (((code[i] + key) - 65) % 26) + 65);
                else
                    printf("%c", code[i]);
            }
            printf("\n");
            return 0;
    }
 }